x/rewards/keeper/flat_fee.go

```diff
@@ -2,28 +2,32 @@ package keeper
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkErrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 	"github.com/archway-network/archway/x/rewards/types"
 )
 
 // SetFlatFee checks if a contract has metadata set and stores the given flat fee to be associated with that contract
-func (k Keeper) SetFlatFee(ctx sdk.Context, contractAddr sdk.AccAddress, flatFee sdk.Coin) error {
+func (k Keeper) SetFlatFee(ctx sdk.Context, senderAddr string, feeUpdate types.FlatFee) error {
```

Comment:
also here we're using senderAddr string for efficiency reason I guess, but can we update the function signature to use sdk.AccAddress? I'd rather be constant with the use of `sdk.AccAddress`.